src/ape_geth/providers.py

```diff
@@ -172,6 +172,17 @@ def connect(self):
             if not self._web3.isConnected():
                 self._geth.disconnect()
                 raise ConnectionError("Unable to connect to locally running geth.")
+        else:
+            # Verify is actually a geth provider
+            client_version = self._web3.clientVersion
+
+            if "geth" not in client_version.lower():
+                network_name = client_version.split("/")[0]
+                raise ProviderError(
+                    f"Attempted to connect geth plugin to non-geth network ('{network_name}')."
+                )
```

Comment:
Can this be a warning instead? I feel like it shouldn't outright fail in case of using an alternative node implementation like Erigon or OpenEthereum